tests/test_runner/test_discover_runner.py

```diff
@@ -122,6 +122,13 @@ def test_name_patterns(self):
                 ).build_suite(['test_runner_apps.simple'])
                 self.assertEqual(expected, self.get_test_methods_names(suite))
 
+    def test_loader_pattern_not_mutated(self):
+        runner = DiscoverRunner(test_name_patterns=['*_tests.py'], verbosity=0)
+        runner.test_loader.testNamePatterns = ['UnittestCase1']
+        count = runner.build_suite(['test_runner_apps.sample']).countTestCases()
+        self.assertEqual(count, 1)
+        self.assertEqual(runner.test_loader.testNamePatterns, ['UnittestCase1'])
```

Comment:
It looks like this test itself isn't isolated. 😄  A context manager like this could be useful for both tests:

```python
@contextmanager
def _temp_loader_patterns(self, patterns):
    original_patterns = DiscoverRunner.test_loader.testNamePatterns
    DiscoverRunner.test_loader.testNamePatterns = patterns
    try:
        yield
    finally:
        DiscoverRunner.test_loader.testNamePatterns = original_patterns
```

For this `patterns` argument, you will want to pass something _different_ from `['UnittestCase1']`, and the assertion that it's not mutated can go in the body of the test itself.